### PR TITLE
fix error using date based index on elasticsearch

### LINKED ISSUE #180: Elasticsearch indice with date error
Hi,

I'm trying to use logagent (docker) to send logs to elasticsearch date based index

My docker command:
```
docker run -d --restart=always \
--name logagent \
--add-host logger-db:172.17.0.1 \
-e LOGS_TOKEN=test-YYYY-MM-DD \
-e LOGS_RECEIVER_URL="http://logger-db:9200" \
-v /var/run/docker.sock:/var/run/docker.sock \
sematext/logagent
```
logger-db is my elasticsearch domain.

Logagent is throwing the error below:
```
Content of /etc/sematext/logagent.conf:
options:
  printStats: 60
  suppress: true
  geoipEnabled: true
  diskBufferDir: /tmp/sematext-logagent

parser:
  patternFiles:
    - /etc/logagent/patterns.yml
output:
  logsene:
    module: elasticsearch
    url: http://logger-db:9200
    index: test-YYYY-MM-DD
logagent -c /etc/sematext/logagent.conf --docker /var/run/docker.sock 
2020-02-04T01:35:47.539Z v3.0.6 pid[6] init filter: inputFilter
2020-02-04T01:35:47.540Z v3.0.6 pid[6] init filter: outputFilter
2020-02-04T01:35:47.541Z v3.0.6 pid[6] set cfg to debug=false for docker-enrichment
2020-02-04T01:35:47.541Z v3.0.6 pid[6] load outputFilter: 0 docker-enrichment
2020-02-04T01:35:47.541Z v3.0.6 pid[6] init plugins
2020-02-04T01:35:47.541Z v3.0.6 pid[6] ../lib/plugins/input/stdin
2020-02-04T01:35:47.557Z v3.0.6 pid[6] ../lib/plugins/output/stdout
2020-02-04T01:35:47.566Z v3.0.6 pid[6] ../lib/plugins/output/elasticsearch.js
2020-02-04T01:35:47.573Z v3.0.6 pid[6] docker-logs
2020-02-04T01:35:47.714Z v3.0.6 pid[6] Logagent report: pid[6] 472 ms 0 lines parsed.  0 lines/s 0.000 MB/s - empty lines: 0
2020-02-04T01:35:47.714Z v3.0.6 pid[6] Logagent stats: usedTokens=0 shippedLogs=0 httpFailed=0 httpRetransmit=0 throughputLinesPerSecond=0
2020-02-04T01:35:47.714Z v3.0.6 pid[6] Memory stats:  heapUsedMB=13 heapTotalMB=29 memoryRssMB=51
2020-02-04T01:35:47.715Z v3.0.6 pid[6] Logagent initialized
2020-02-04T01:35:47.728Z v3.0.6 pid[6] Container 4d88acce73bd024a8abdd3596464655d1b151e105eaafcb3d83cd80db4fdc03f /logger-db setting LOGSENE_ENABLED=true
2020-02-04T01:35:47.731Z v3.0.6 pid[6] Container 6efc8c9b9eaafd457d7d539faadb305169f2c81c1238d3b6f5df5cc145ed581b /logger-ui setting LOGSENE_ENABLED=true
using type logs
2020-02-04T01:36:47.715Z v3.0.6 pid[6] Logagent report: pid[6] 60000 ms 3 lines parsed.  0 lines/s 0.000 MB/s - empty lines: 0
2020-02-04T01:36:47.715Z v3.0.6 pid[6] Logagent stats: usedTokens=1 shippedLogs=0 httpFailed=0 httpRetransmit=0 throughputLinesPerSecond=0
2020-02-04T01:36:47.715Z v3.0.6 pid[6] Memory stats:  heapUsedMB=12 heapTotalMB=14 memoryRssMB=52
2020-02-04T01:36:48.129Z v3.0.6 pid[6] Error in Elasticsearch request:  source="logsene-js" err={"message":"HTTP status code:400 Error: {\"type\":\"invalid_index_name_exception\",\"reason\":\"Invalid index name [test-YYYY-MM-DD], must be lowercase\",\"index_uuid\":\"_na_\",\"index\":\"test-YYYY-MM-DD\"}","httpStatus":400,"httpBody":{"_index":"test-YYYY-MM-DD","_type":"logs","_id":null,"status":400,"error":{"type":"invalid_index_name_exception","reason":"Invalid index name [test-YYYY-MM-DD], must be lowercase","index_uuid":"_na_","index":"test-YYYY-MM-DD"}},"url":"http://logger-db:9200/test-2020-02-04/_bulk"} /  message="HTTP status code:400 Error: {\"type\":\"invalid_index_name_exception\",\"reason\":\"Invalid index name [test-YYYY-MM-DD], must be lowercase\",\"index_uuid\":\"_na_\",\"index\":\"test-YYYY-MM-DD\"}" httpStatus=400 httpBody={"_index":"test-YYYY-MM-DD","_type":"logs","_id":null,"status":400,"error":{"type":"invalid_index_name_exception","reason":"Invalid index name [test-YYYY-MM-DD], must be lowercase","index_uuid":"_na_","index":"test-YYYY-MM-DD"}} url="http://logger-db:9200/test-2020-02-04/_bulk"
2020-02-04T01:36:48.129Z v3.0.6 pid[6] Error in Elasticsearch request:  source="logsene-js" err={"message":"HTTP status code:400 Error: {\"type\":\"invalid_index_name_exception\",\"reason\":\"Invalid index name [test-YYYY-MM-DD], must be lowercase\",\"index_uuid\":\"_na_\",\"index\":\"test-YYYY-MM-DD\"}","httpStatus":400,"httpBody":{"_index":"test-YYYY-MM-DD","_type":"logs","_id":null,"status":400,"error":{"type":"invalid_index_name_exception","reason":"Invalid index name [test-YYYY-MM-DD], must be lowercase","index_uuid":"_na_","index":"test-YYYY-MM-DD"}},"url":"http://logger-db:9200/test-2020-02-04/_bulk"} /  message="HTTP status code:400 Error: {\"type\":\"invalid_index_name_exception\",\"reason\":\"Invalid index name [test-YYYY-MM-DD], must be lowercase\",\"index_uuid\":\"_na_\",\"index\":\"test-YYYY-MM-DD\"}" httpStatus=400 httpBody={"_index":"test-YYYY-MM-DD","_type":"logs","_id":null,"status":400,"error":{"type":"invalid_index_name_exception","reason":"Invalid index name [test-YYYY-MM-DD], must be lowercase","index_uuid":"_na_","index":"test-YYYY-MM-DD"}} url="http://logger-db:9200/test-2020-02-04/_bulk"
2020-02-04T01:36:48.129Z v3.0.6 pid[6] Error in Elasticsearch request:  source="logsene-js" err={"message":"HTTP status code:400 Error: {\"type\":\"invalid_index_name_exception\",\"reason\":\"Invalid index name [test-YYYY-MM-DD], must be lowercase\",\"index_uuid\":\"_na_\",\"index\":\"test-YYYY-MM-DD\"}","httpStatus":400,"httpBody":{"_index":"test-YYYY-MM-DD","_type":"logs","_id":null,"status":400,"error":{"type":"invalid_index_name_exception","reason":"Invalid index name [test-YYYY-MM-DD], must be lowercase","index_uuid":"_na_","index":"test-YYYY-MM-DD"}},"url":"http://logger-db:9200/test-2020-02-04/_bulk"} /  message="HTTP status code:400 Error: {\"type\":\"invalid_index_name_exception\",\"reason\":\"Invalid index name [test-YYYY-MM-DD], must be lowercase\",\"index_uuid\":\"_na_\",\"index\":\"test-YYYY-MM-DD\"}" httpStatus=400 httpBody={"_index":"test-YYYY-MM-DD","_type":"logs","_id":null,"status":400,"error":{"type":"invalid_index_name_exception","reason":"Invalid index name [test-YYYY-MM-DD], must be lowercase","index_uuid":"_na_","index":"test-YYYY-MM-DD"}} url="http://logger-db:9200/test-2020-02-04/_bulk"
```
It looks like date in index name is replaced correctly for url but it is not replaced in request body.

I have checked docker versions:
latest
3.0.6
3.0.5
3.0.3

Is it something that I'm doing wrong?

### PR DESCRIPTION
**Pull request by aya**
Thanks to @comel, this PR fixes #180 Elasticsearch indice with date error

### CODE REVIEW

File: lib/plugins/output/elasticsearch.js [hunk 1]
```diff
@@ -194,12 +194,18 @@ OutputElasticsearch.prototype.eventHandler = function (data, context) {
 
   if (index) {
     index = applyDateFormatToIndex(index, data)
+    if (data._index != undefined) {
```
Comment (adnanrahic) [inline lib/plugins/output/elasticsearch.js:197]: To align with our linting rules, can you edit this to use strict inequality?

```suggestion
    if (data._index !== undefined) {
```

File: lib/plugins/output/elasticsearch.js [hunk 2]
```diff
@@ -194,12 +194,18 @@ OutputElasticsearch.prototype.eventHandler = function (data, context) {
...
+      data._index = index
+    }
     this.indexData(index, data._type || 'logs', data, config, logsReceiverUrl)
   }
 
   if (context.logsReceivers && context.logsReceivers.length > 0) {
     context.logsReceivers.forEach(receiver => {
       const receiverIndex = applyDateFormatToIndex(receiver.index, data)
+      if (data._index != undefined) {
```
Comment (adnanrahic) [inline lib/plugins/output/elasticsearch.js:206]: To align with our linting rules, can you edit this to use strict inequality?

```suggestion
      if (data._index !== undefined) {
```

**Review by adnanrahic — CHANGES_REQUESTED** [review]
Thanks a lot for this!
There is only one tiny linting edit I'd really like to see.
Then we can go ahead and merge it!

**Comment by aya** [comment]
I applied you suggestions in latest commit, thanks for review !

**Review by adnanrahic — APPROVED** [review]

**Comment by adnanrahic** [comment]
Sweet! Thanks for the contribution. I'll merge this now.

**Pull request merged**